datasets/seq_cifar100_label_drift.py

```diff
@@ -0,0 +1,237 @@
+            self.classes = []
+            return
+
+        mask = np.zeros_like(np.array(self.targets))
+        for label in classes_list:
+            mask = np.logical_or(mask, np.array(self.targets) == label)
+        self.data = self.data[mask]
+        self.targets = np.array(self.targets)[mask]
+
+        self.classes = classes_list
+
+    def apply_drift(self, classes: list):
+        if len(set(self.classes).union(classes)) == 0:
+            return
+        self.drifted_classes.extend(classes)
+
+        # switch meta-class in target_transform dictionary
+        for c in classes:
+            s = self.superclass_mapping[c]
+            self.metaclass_mapping[s] = 0 if self.metaclass_mapping[s] == 1 else 1
+
+    def prepare_normal_data(self):
+        pass
+
+
+class TestCIFAR100LabelDrift(MammothDataset, CIFAR100):
+    """Workaround to avoid printing the already downloaded messages."""
+
+    def __init__(self, root, transform, target_transform):
+        self.root = root
+        super().__init__(
+            root,
+            train=False,
+            transform=transform,
+            target_transform=target_transform,
+            download=not self._check_integrity(),
+        )
+        self.superclass_mapping = target_transform.superclass_mapping
+        self.metaclass_mapping = target_transform.metaclass_mapping
+        self.classes = list(range(100))
+
+    def __getitem__(self, index: int) -> Tuple[Image.Image, int]:
+        """
+        Args: index (int): Index
+        Returns: tuple: (image, target) where target is index of the target class.
+        """
+        img, target = self.data[index], self.targets[index]
+
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        img = Image.fromarray(img)
+        img = self.transform(img)
+        target = self.superclass_mapping[target]
+        target = self.metaclass_mapping[target]
+
+        return img, target
+
+    def select_classes(self, classes_list: list[int]):
+        if len(classes_list) == 0:
+            self.data = np.array([])
+            self.targets = np.array([])
+            self.classes = []
+            return
+
+        mask = np.zeros_like(np.array(self.targets))
+        for label in classes_list:
+            mask = np.logical_or(mask, np.array(self.targets) == label)
+        self.data = self.data[mask]
+        self.targets = np.array(self.targets)[mask]
+
+        self.classes = classes_list
+
+    def apply_drift(self, classes: list):
+        if len(set(self.classes).union(classes)) == 0:
+            return
+        self.drifted_classes.extend(classes)
+
+        # switch meta-class in target_transform dictionary
+        for c in classes:
+            s = self.superclass_mapping[c]
+            self.metaclass_mapping[s] = 0 if self.metaclass_mapping[s] == 1 else 1
+
+    def prepare_normal_data(self):
+        pass
+
+
+class TargetTransform:
```

Comment:
It seems that this class is not needed. You can simply pass superclass_target_mapping and metaclass_target_mapping directly to the TrainCIFAR100LabelDrift constructor instead of single target_transform parameter. I would be simpler.